labscript/labscript.py

```diff
@@ -30,2717 +29,85 @@
-        if self._static_value is None:
-            if not config.suppress_mild_warnings and not config.suppress_all_warnings:
-                sys.stderr.write(' '.join(['WARNING:', self.name, 'has no value set. It will be set to %s.\n'%self.instruction_to_string(self.default_value)]))
-            self._static_value = self.default_value
-        return self._static_value
-    
-
-class StaticDigitalOut(StaticDigitalQuantity):
-    """Static Digital Output class for use with all devices that have constant outputs."""
-    description = 'static digital output'
-        
-class AnalogIn(Device):
-    """Analog Input for use with all devices that have an analog input."""
-    description = 'Analog Input'
-    
-    @set_passed_properties(property_names = {})
-    def __init__(self,name,parent_device,connection,scale_factor=1.0,units='Volts',**kwargs):
-        """Instantiates an Analog Input.
-
-        Args:
-            name (str): python variable to assign this input to.
-            parent_device (:obj:`IntermediateDevice`): Device input is connected to.
-            scale_factor (float, optional): Factor to scale the recorded values by.
-            units (str, optional): Units of the input.
-            **kwargs: Keyword arguments passed to :func:`Device.__init__`.
-        """
-        self.acquisitions = []
-        self.scale_factor = scale_factor
-        self.units=units
-        Device.__init__(self,name,parent_device,connection, **kwargs)
-   
-    def acquire(self,label,start_time,end_time,wait_label='',scale_factor=None,units=None):
-        """Command an acquisition for this input.
-
-        Args:
-            label (str): Unique label for the acquisition. Used to identify the saved trace.
-            start_time (float): Time, in seconds, when the acquisition should start.
-            end_time (float): Time, in seconds, when the acquisition should end.
-            wait_label (str, optional): 
-            scale_factor (float): Factor to scale the saved values by.
-            units: Units of the input, consistent with the unit conversion class.
-
-        Returns:
-            float: Duration of the acquistion, equivalent to `end_time - start_time`.
-        """
-        if scale_factor is None:
-            scale_factor = self.scale_factor
-        if units is None:
-            units = self.units
-        self.acquisitions.append({'start_time': start_time, 'end_time': end_time,
-                                 'label': label, 'wait_label':wait_label, 'scale_factor':scale_factor,'units':units})
-        return end_time - start_time
+# This imports the default Qt library that other labscript suite code will
+# import as well, since it all uses qtutils. By having a Qt library already
+# imported, we ensure matplotlib (imported by pylab) will notice this and use
+# the same Qt library and API version, and therefore not conflict with any
+# other code is using:
+import qtutils
 
+from pylab import *
```

Comment:
Maybe this is one step too far, but I would love to be able to remove this star import as part of this. Maybe that's a bad idea given that current docs recommend `from labscript import *` so downstream users are likely accidentally using imports from pylab so it would be much more breaking than this PR is intending...

But really, pylab use is apparently strongly discouraged and it really pollutes the `labscript` namespace with a ton of stuff that is never going to be needed (a whopping 925 entries, most being plotting utilities that scripts really shouldn't be using inadvertently anyway).

Reply:
This is definitely a future goal! I'd like to try and limit the scope creep, especially as removing that import would mean a major version bump

Reply:
Totally fair. I'll go ahead and make some issues for this and the other scope creep to help me remember they are on the TODO list.